src/GovUk.Education.ExploreEducationStatistics.Data.Api.Tests/Services/PermalinkServiceTests.cs

```diff
@@ -0,0 +1,658 @@
+                // I.e. when field 'Locations' is still defined in ResultSubjectMetaViewModel
+                // TODO EES-2902 No need to do this after the Locations field is dropped from the type.
+                existingLegacyLocationsJsonArray.Replace(legacyLocationsJsonArray);
+            }
+            else
+            {
+                subjectMetaJsonObject!.Add("Locations", legacyLocationsJsonArray);
+            }
+
+            var blobStorageService = new Mock<IBlobStorageService>(MockBehavior.Strict);
+            var subjectRepository = new Mock<ISubjectRepository>(MockBehavior.Strict);
+
+            blobStorageService.SetupDownloadBlobText(
+                container: Permalinks,
+                path: permalink.Id.ToString(),
+                blobText: permalinkJsonObject.ToString(Formatting.None));
+
+            subjectRepository
+                .Setup(s => s.Get(subject.Id))
+                .ReturnsAsync(subject);
+
+            subjectRepository
+                .Setup(s => s.IsSubjectForLatestPublishedRelease(subject.Id))
+                .ReturnsAsync(true);
+
+            subjectRepository
+                .Setup(s => s.FindPublicationIdForSubject(subject.Id))
+                .ReturnsAsync(_publicationId);
+
+            var service = BuildService(blobStorageService: blobStorageService.Object,
+                subjectRepository: subjectRepository.Object);
+
+            var result = (await service.Get(permalink.Id)).AssertRight();
+
+            MockUtils.VerifyAllMocks(
+                blobStorageService,
+                subjectRepository);
+
+            Assert.Equal(permalink.Id, result.Id);
+
+            var subjectMeta = result.FullTable.SubjectMeta;
+
+            // Expect Locations to be empty
+            Assert.Empty(subjectMeta.Locations);
+
+            // Expect Locations to have been transformed to LocationsHierarchical
+            Assert.Single(subjectMeta.LocationsHierarchical);
+            Assert.True(subjectMeta.LocationsHierarchical.ContainsKey("localAuthority"));
+
+            var localAuthorities = subjectMeta.LocationsHierarchical["localAuthority"];
+            Assert.Equal(3, localAuthorities.Count);
+
+            Assert.Equal(legacyLocations[0].Label, localAuthorities[0].Label);
+            Assert.Equal(legacyLocations[0].Value, localAuthorities[0].Value);
+            Assert.Equal(legacyLocations[0].GeoJson, localAuthorities[0].GeoJson);
+
+            Assert.Equal(legacyLocations[1].Label, localAuthorities[1].Label);
+            Assert.Equal(legacyLocations[1].Value, localAuthorities[1].Value);
+            Assert.Equal(legacyLocations[1].GeoJson, localAuthorities[1].GeoJson);
+
+            Assert.Equal(legacyLocations[2].Label, localAuthorities[2].Label);
+            Assert.Equal(legacyLocations[2].Value, localAuthorities[2].Value);
+            Assert.Equal(legacyLocations[2].GeoJson, localAuthorities[2].GeoJson);
+        }
+
+        [Fact]
```

Comment:
I guess we're never planning on running with the flag enabled for a time and then switching back to it being disabled in an emergency?  Otherwise we'd need a converter to also transform `LocationsHierarchical` back to `Locations` as well.

Reply:
Good point thanks, but not really planning for that to happen.

We only plan to turn the feature on for local development while the FE ticket EES-2777 is in progress. 

The toggle exists to prevent a long running code branch while these changes are in progress and not designed to be toggled in real environments as the BE change it brings can't  work independently without the FE change EES-2777.

I don't think we plan to keep the toggle either once the FE changes are complete so we should never have a situation in a real environment where `LocationsHierarchical` need transforming back to `Locations`.

Reply:
Yup no worries - I figured so, thanks!